src/main/java/gregtech/common/covers/filter/OreDictionaryItemFilter.java

```diff
@@ -123,12 +127,43 @@ public Object matchItemStack(ItemStack itemStack) {
 
     public boolean matchesItemStack(ItemStack itemStack) {
         if (this.error) return false;
-        Boolean cached = this.matchCache.get(itemStack);
-        if (cached != null) {
+        Item item = itemStack.getItem();
+        ItemVariantMap<Set<String>> oreDictEntry = OreDictUnifier.getOreDictionaryEntry(item);
+
+        if (oreDictEntry == null) {
+            // no oredict entries associated
+            Boolean cached = this.noOreDictMatch.getEntry();
+            if (cached == null) {
+                cached = this.glob.matches("");
+            }
+            this.matchCache.put(item, this.noOreDictMatch);
             return cached;
         }
+
+        ItemVariantMap.Mutable<Boolean> cacheEntry = this.matchCache.get(item);
+        if (cacheEntry != null) {
+            Boolean cached = cacheEntry.getEntry(itemStack);
+            if (cached != null) return cached;
+        }
+
+        if (cacheEntry == null) {
+            if (oreDictEntry.isEmpty()) {
+                // no oredict entries associated
+                Boolean cached = this.noOreDictMatch.getEntry();
+                if (cached == null) {
+                    this.noOreDictMatch.setEntry(cached = this.glob.matches(""));
```

Comment:
Given that this input may be supplied often, would it be a good idea to add a special case in the glob implementations for early returns if the glob matches empty stings?

Also, using the result of variable assignment in a method call is somewhat confusing. It would be easy to interpret incorrectly, so it would be better to assign the value to `cached` outside of the `setEntry()` call, then pass in that value afterwards.